session.go

```diff
@@ -222,6 +218,9 @@ func (s *Session) NewReceiver(ctx context.Context, source string, opts *Receiver
 
 // NewSender opens a new sender link on the session.
 // opts: pass nil to accept the default values.
+//
+// If the context is cancelled or its deadline expires, the
+// underlying Conn is closed and a *ConnError is returned.
```

Comment:
It might make more sense to just kill the session only at this point since we know it was created. We're just saying the session is possibly no longer a valid container.

Reply:
Killing the session requires a handshake between peers.  If the context has already expired, how long do we wait for that handshake to complete?

Reply:
> We're just saying the session is possibly no longer a valid container.

I don't think we can make this assumption.  We sent an attach performative but didn't receive the ack in the time allotted.
- the peer is under load, and we just didn't wait long enough
- we have a bug somewhere and the frame was lost along the way

In both cases, we (and our peer) have allocated a handle for the link.  It's not safe to reuse the handle as it's associated with a link (we'd have to close the link first).

Reply:
Thinking about this more, even if we were to kill the session, this can also introduce a race.  If there was a delay in the peer sending the attach ack, and we receive it after the session has been killed, this will result in the connection being terminated as we've now received a frame for a channel that no longer exists.